dbms/src/Storages/Transaction/ProxyFFIStatusService.cpp

```diff
@@ -52,14 +52,31 @@ HttpRequestRes HandleHttpRequestSyncStatus(
     std::vector<RegionID> region_list;
     size_t count = 0;
 
+    const size_t max_print_region = 30;
     // if storage is not created in ch, flash replica should not be available.
     if (tmt.getStorages().get(table_id))
     {
-        tmt.getRegionTable().handleInternalRegionsByTable(table_id, [&](const RegionTable::InternalRegions & regions) {
-            count = regions.size();
+        RegionTable & region_table = tmt.getRegionTable();
+        region_table.handleInternalRegionsByTable(table_id, [&](const RegionTable::InternalRegions & regions) {
             region_list.reserve(regions.size());
+            bool can_log = region_table.canPrintLog();
+            String lag_regions_log;
+            size_t print_count = 0;
             for (const auto & region : regions)
-                region_list.push_back(region.first);
+            {
+                UInt64 leader_safe_ts;
+                UInt64 self_safe_ts;
+                if (!region_table.isSafeTSLag(region.first, &leader_safe_ts, &self_safe_ts))
+                {
+                    region_list.push_back(region.first);
+                } else if (can_log && print_count < max_print_region)
+                {
+                        lag_regions_log += fmt::format("lag_region_id: {}, leader_safe_ts: {}, self_safe_ts: {}; ", region.first, leader_safe_ts, self_safe_ts);
+                        print_count++;
+                }
+            }
+            count = region_list.size();
+            LOG_FMT_DEBUG(&Poco::Logger::get(__FUNCTION__), "table_id={}, total_region_count={}, ready_region_count={}, lag_region_info={}", table_id, regions.size(), count, lag_regions_log);
```

Comment:
show the number of **all** lag regions

Reply:
> show the number of **all** lag regions

Has showed the number of total region count and ready region count.